utils/es.py

```diff
@@ -0,0 +1,19 @@
+from elasticsearch import Elasticsearch as Elasticsearch_official
+from elasticsearch import __version__ as es_pylib_version
+
+
+class Elasticsearch(Elasticsearch_official):
+    """Elasticsearch wrapper class
+
+    Wrapper class around the official Elasticsearch class that adds a simple version check upon initialization. In particular it checks if the major version of the library in use match the one of the cluster that we are tring to interact with. The check can be skip by setting to false the check_version parameter.
+    """
+
+    def __init__(self, check_version=True, *args, **kargs):
+        super(Elasticsearch, self).__init__(*args, **kargs)
+        if check_version:
+            es_version = self.get_version()
+            if(int(es_version[0]) != int(es_pylib_version[0])):
+                raise AssertionError("The Elasticsearch python library version does not match the one of the running cluster: {} != {}. Please install the correct elasticsearch-py version".format(es_pylib_version[0], es_version[0]))
```

Comment:
I would prefer not to do this. I think that a good solution might be the use of a "factory" function: that is, something that, using the exactly same logic of this code
* instantiate the official Elasticsearch class
* check if everything is right and either:
** raise an Exception (see below)
** returns the unmodified object

Subclassing should be used only when we really mean it.

## On AssertionError

btw, I don't think AssertionError is appropriate here; [doc](https://docs.python.org/3.5/library/exceptions.html#AssertionError) is clear that is reserved to the `assert` statement, which is not what we're doing.  As an example, `AssertionError`s would be surprising when running with `python -O`. I think that a `RuntimeError` can be good enough for our case

Reply:
I thought on this, the thing that I like of this approach is that it is very easy to revert. As you can see I needed to just change the import line in all the files that use the Elasticsearch class, but I didn't actually change any code.

About the AssertionError I didn't know that the python optimization skip those. I've change it to RuntimeError as you suggested.

Reply:
> I thought on this, the thing that I like of this approach is that it is very easy to revert. As you can see I needed to just change the import line in all the files that use the Elasticsearch class, but I didn't actually change any code.

same applies if you change `class Elasticsearch` to `def Elasticsearch`. You can still keep code compatibility, returning an object that is the "true" one (in case of success).

> About the AssertionError I didn't know that the python optimization skip those.

That's not what I meant. Python optimization skips `asserts`; it doesn't ignore `AssertionError`s if those are raised by user code.
But what you're doing here is not an assert. Generally speaking, when an assert is violated it means that the programmer committed an error; that's why assert can be considered a debugging tool, or even a "documentation" tool, not an error-checking tool.
An assert should never check user input, system environment, or anything that is out of the control of the programmer.
What I meant is that raising an `AssertionError` would be confusing, because those are _unexpected_ with `python -O`. But this is just one of the consequences of using `AssertionError` for something that is not an `assert` :)